force-app/main/default/classes/P0.cls

```diff
@@ -0,0 +1,81 @@
+public with sharing class P0 {
+    
+    // Return the nth fiboacci number in the fibonacci sequence.
+    public static integer Fibonacci(integer n) { 
+        integer num1 = 1;
+        integer num2 = 1;
+        for (integer i=2; i<n; i++) {
+            integer fibo = num2 + num1;
+            num1 = num2;
+            num2 = fibo;
+        }
+        return num2;
+    }
+
+    //Given a string, return its backwards order.
+    public static String reverseString(String s) {
+        integer stringLength = s.length();
+        String[] arrayOfString = new List<String>();
+        for(integer i=stringLength-1; i>=0; i--) {
+            arrayOfString.add(s.substring(i,i+1));
+        }
+        s = String.join(arrayOfString,'');
+        return s;
+    }
+
+    //Given a string, determine if it is a palindrome
+    public static Boolean isPalindrome(String s) {
+        if(P0.reverseString(s)==s)
+        return true;
+        else {
+            return false;
+        }
+    }
+
+    //computes the factorial of a given number using recursion.
+    public static Integer factorial(Integer n) {
+        if(n>=1){
+            return n*factorial(n-1);
+        }
+        else {
+            return 1;
+        }
+    }
+
+    //Given a state, return the accounts in that state
+    public static List<Account> getAccountsForAGivenState(State s) {
+       Account[] acc = [SELECT Account.Name, ( SELECT FirstName, LastName FROM Contacts WHERE MailingState=:s.name() ) FROM Account WHERE BillingState=:s.name()];
+        return acc;
+    }
+
+
+    //For each student, get the grade for each of their classes
+    public static Map<Id, Map<Id, Double>> getClassGrades() {
+        
+        List<ClassEnrollment__c> classenrollments = [SELECT ClassEnrollment__c.Name, ClassEnrollment__c.Id,ClassEnrollment__c.Class__r.Id, ClassEnrollment__c.Student__r.Name, ClassEnrollment__c.Student__r.Id FROM ClassEnrollment__c];// 
```

Comment:
SInce you're querying the CE object, there's not need to do ClassEnrollment__c.Name, and etc. Just use Name, Id, Class__r.Id,...